--- svdrp.c.orig	2024-04-22 15:05:01.634228904 +0200
+++ svdrp.c	2024-04-22 15:12:23.447578886 +0200
@@ -905,6 +905,10 @@
   "MESG <message>\n"
   "    Displays the given message on the OSD. The message will be queued\n"
   "    and displayed whenever this is suitable.\n",
+  "MSGT <message>\n"
+  "    Displays the given message on the OSD. First letter is used to get the\n"
+  "    message type. Default is info. Use '%' for warning and '@' for error.\n"
+  "    The message will be queued and displayed whenever this is suitable.\n",
   "MODC <number> <settings>\n"
   "    Modify a channel. Settings must be in the same format as returned\n"
   "    by the LSTC command.",
@@ -1087,7 +1091,8 @@
   void CmdLSTE(const char *Option);
   void CmdLSTR(const char *Option);
   void CmdLSTT(const char *Option);
-  void CmdMESG(const char *Option);
+  void CmdMESG(const char *Option);  // Message (Info)
+  void CmdMSGT(const char *Option);  // Mesage with types
   void CmdMODC(const char *Option);
   void CmdMODT(const char *Option);
   void CmdMOVC(const char *Option);
@@ -1998,6 +2003,23 @@
      Reply(501, "Missing message");
 }
 
+void cSVDRPServer::CmdMSGT(const char *Option)
+{
+  if (*Option) {
+     eMessageType Type = mtInfo;              // Default
+     const char First = Option[0];            // First character
+     if (First == '%') Type = mtWarning;      // Percent (#37)
+     else if (First == '@') Type = mtError;   // At (#64)
+
+     if (!(Type == mtInfo)) Option++;  // Move the pointer one to the right
+     isyslog("SVDRP %s < %s message '%s'", Setup.SVDRPHostName, *clientName, Option);
+     Skins.QueueMessage(Type, Option);
+     Reply(250, "message queued");
+     }
+  else
+     Reply(501, "Missing message");
+}
+
 void cSVDRPServer::CmdMODC(const char *Option)
 {
   if (*Option) {
@@ -2590,7 +2612,8 @@
   else if (CMD("LSTE"))  CmdLSTE(s);
   else if (CMD("LSTR"))  CmdLSTR(s);
   else if (CMD("LSTT"))  CmdLSTT(s);
-  else if (CMD("MESG"))  CmdMESG(s);
+  else if (CMD("MESG"))  CmdMESG(s);  // Message (Info)
+  else if (CMD("MSGT"))  CmdMSGT(s);  // Mesage with types
   else if (CMD("MODC"))  CmdMODC(s);
   else if (CMD("MODT"))  CmdMODT(s);
   else if (CMD("MOVC"))  CmdMOVC(s);
